File: python-code/algorithms/sort/insertion_sort.py
def insertion_sort(nums):
    for i in range(1, len(nums)):
        target = nums[i]
        j = i - 1
        while target < nums[j]:
            nums[j + 1] = nums[j]
            j -= 1
        nums[j + 1] = target


nums = [5, 2, 7, 3, 8, 3, 4, 5, 6, 8, 2, 4]
insertion_sort(nums)

File: java-code/src/main/java/me/joohyuk/basic/sort/InsertionSort.java
package me.joohyuk.basic.sort;

public class InsertionSort {

    public static void insertionSort(int[] nums) {
        for (int i = 1; i < nums.length; i++) {
            int target = nums[i];
            int j = i - 1;

            while (j >= 0 && target < nums[j]) {
                nums[j + 1] = nums[j];
                j -= 1;
            }

            nums[j + 1] = target;
        }
    }

    public static void main(String[] args) {
        int[] nums = {5, 2, 3, 4, 65, 6, 45, 3, 657, 456, 7, 456, 7};
        insertionSort(nums);
        for (int num : nums) {
            System.out.print(num + " ");
        }
    }
}
